client/client.go

```diff
@@ -217,6 +217,40 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// VerifyInclusion ensures that the given leaf data has been included in the log.
+func (c *LogClient) VerifyInclusion(ctx context.Context, data []byte) error {
+	leaf := c.buildLeaf(data)
+	if err := c.UpdateRoot(ctx); err != nil {
+		return fmt.Errorf("UpdateRoot(): %v", err)
+	}
+	return c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
+}
+
+// VerifyInclusionAtIndex ensures that the given leaf data has been included in the log at a particular index.
```

Comment:
Added documentation.

I personally find it easier to reason about the state of the client if the client *always* first updates the root, otherwise "not found" failures are ambiguous. Did we request an inclusion proof to too small of a tree size, or is the requested item not there? 

Is there a better, simpler algorithm for interacting with the log?